src/plugins/discover/public/application/components/utils/use_pagination.ts

```diff
@@ -4,13 +4,25 @@
  */
 
 import { useState, useMemo, useCallback } from 'react';
+import { DiscoverServices } from '../../../build_services';
+import { SAMPLE_SIZE_SETTING } from '../../../../common';
+import { generatePageSizeOptions } from './page_size_options';
+export interface Props {
+  services: DiscoverServices;
+  rowCount: number;
+}
+
+export const usePagination = ({ rowCount, services }: Props) => {
```

Comment:
I'd avoid passing the entire `services` object, but passing `pageSizeLimit` here which is `sampleSize` in this case. Does it make sense?

```suggestion
export const usePagination = (rowCount: number, pageSizeLimit: number) => {
```

Reply:
yes, thanks for the suggestions.